src/features/contactmoment/ContactverzoekenOverzicht.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div class="container">
+    <div class="header">
+      <span>Datum</span>
+      <span>Status</span>
+      <span>Behandelaar</span>
+      <span>Afgerond</span>
+      <span class="chevron"></span>
+    </div>
+
+    <div
+      v-for="(contactverzoek, idx) in contactverzoeken"
+      :key="idx"
+      class="verzoek-item"
+    >
+      <button
+        @click="toggleItemContent(idx)"
+        class="verzoek-item-header"
+        :class="{ 'is-active': activeContactverzoeken[idx] }"
+      >
+        <span>{{ contactverzoek.datum }}</span>
+        <span>{{ contactverzoek.status }}</span>
+        <span>{{ contactverzoek.behandelaar }}</span>
+        <span>{{ contactverzoek.afgerond }}</span>
+        <span class="chevron icon-after chevron-down"></span>
+      </button>
+
+      <div
+        class="verzoek-item-content"
+        :class="{ 'is-active': activeContactverzoeken[idx] }"
+      >
+        <dl>
+          <dt>Starttijd</dt>
+          <dd>{{ contactverzoek.starttijd }}</dd>
+        </dl>
+
+        <dl>
+          <dt>Aanmaker</dt>
+          <dd>{{ contactverzoek.aanmaker }}</dd>
+        </dl>
+
+        <dl>
+          <dt>Status</dt>
+          <dd>{{ contactverzoek.status }}</dd>
+        </dl>
+
+        <dl>
+          <dt>Vraag</dt>
+          <dd>{{ contactverzoek.vraag }}</dd>
+        </dl>
+
+        <dl>
+          <dt>Notitie</dt>
+          <dd>
+            {{ contactverzoek.notitie }}
+          </dd>
+        </dl>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref } from "vue";
+import type { ContactverzoekDetail } from "./types";
+
+const props = defineProps<{
+  contactverzoeken: ContactverzoekDetail[];
+}>();
+
+const activeContactverzoeken = ref(props.contactverzoeken.map((_) => false));
+
+const toggleItemContent = (idx: number) => {
+  activeContactverzoeken.value[idx] = !activeContactverzoeken.value[idx];
+};
```

Comment:
This works, however, I wanted to hook into the `props.contactverzoeken` here, and add the ref to a mapped object, like so (dummy code):

```
contactverzoeken.map((contactverzoek) => ({ ...contactverzoek, isOpen: false }));
```

Thereafter, we could use that same mapped object for everything. But I couldn't get the `isOpen` to be reactive (tried refs and computed, but somewhere I did something wrong I think).

Current implementation works fine, but might be nice to refactor with above mindset, if you know how to do it.

Reply:
een apart component voor de listitems, die z'n eigen toggle state bijhoudt zou ik een nog mooiere oplossing vinden dan een extra isOpen property toevoegen aan het contactverzoek. Maar wat mij betreft kan je de huidige oplossing gewoon zo laten.